system/alternatives.py

```diff
@@ -114,7 +124,16 @@ def main():
                         link = line.split()[1]
                         break
 
-    if current_path != path:
+    if not force:
+        need_to_force = False
+    elif not os.path.islink(link):
```

Comment:
No, it shouldn't do that and the first branch of the if handles that specifically and with the highest precedence of all the options here.